### ISSUE TITLE
[Code] - DQN, DDQN - Gradient Clipping

### ISSUE THREAD

**Issue by Wei-1** [Wei-1/issues#39]
Based on the read: [Git](https://github.com/Curt-Park/rainbow-is-all-you-need/blob/master/01.dqn.ipynb)

The reason why our DQN and DDQN or other DRL algorithm in Scala-Machine-Learning are not stable in CI is because of the missing of an important concept: Gradient Clipping
Which need to regularize the reward error between -1 and 1.
 - [x] [DQN](https://github.com/Wei-1/Scala-Machine-Learning/blob/develop/src/main/scala/algorithm/reinforcement/DQN.scala#L47)
 - [x] DDQN
   - [x] [Over Optimistics](https://github.com/Wei-1/Scala-Machine-Learning/blob/develop/src/main/scala/algorithm/reinforcement/DDQN.scala#L69)
 - [x] Q-Neural-Learning
 - [x] A3C

After further investigation, I found that we will need to use the updated neural network instead of the Basic-NN to achieve this upgrade.
Therefore, this update should happen after the new NN update and unit tests.
